### PR TITLE
set OTLP service name and namespace

### LINKED ISSUE #180: Zipkin reports UNKNOWN_SERVICE : graphql_request for router spans
**Describe the bug**
As a follow-on to #149 I'm now seeing Zipkin report `UNKNOWN_SERVICE : graphql_request` for router spans.

The subgraph spans are also offset, so not sure if trace propagation is working correctly.

**To Reproduce**
1. `git clone git@github.com:apollographql/supergraph-demo-fed2.git`
2. `make docker-up-router-otel` - uses [docker-compose.router-otel.yml](https://github.com/apollographql/supergraph-demo-fed2/blob/main/docker-compose.router-otel.yml)
3. `make smoke-router`
4. http://localhost:9411/ -> `Run Query`
5. see `UNKNOWN_SERVICE : graphql_request` for router requests
6. click `Show` and see the offset subgraph spans

**Expected behavior**
1. router to set a default resource/service name so it shows up in the Zipkin `serviceName` field, and ideally allow setting the field via `config.yaml` and include it in the [config docs](https://www.apollographql.com/docs/router/configuration/)
2. subgraph trace spans to align with the parent router spans, like it does for the Gateway in https://github.com/apollographql/supergraph-demo-fed2#tracing-with-open-telemetry

**Output**
image

**Desktop (please complete the following information):**
```
ProductName:	macOS
ProductVersion:	11.6
BuildVersion:	20G165
```

### PR DESCRIPTION
**Pull request by Geal**
fix #180
related to #38

- a5b8581 sets default values for service name and namespace, to fix #180
- 26045c3 fixes a regression introduced in https://github.com/apollographql/router/commit/34f5f9ea64fdeb488e0c99a4420394cc066b7744: if reqwest-tracing uses a different tracing-opentelemetry version, tracing headers are not propagated
- 3bc699f fixes a regression introduced in 44840e4: the execute and format_response phases were appearing as their own root spans

26045c3 is temporary, I'll send a PR to reqwest-tracing to support the tracing-opentelemetry 0.16

we should get #38 implemented, to make sure those kinds of regressions do not happen

### CODE REVIEW

File: apollo-router-core/Cargo.toml [hunk 1]
```diff
@@ -35,7 +35,7 @@ mockall = "0.10.2"
 static_assertions = "1"
 test-log = { version = "0.2.8", default-features = false, features = ["trace"] }
 tokio = { version = "1", features = ["full"] }
-tracing-subscriber = { version = "0.3.2", default-features = false, features = [
+tracing-subscriber = { version = "0.2.25", default-features = false, features = [
```
Comment (BrynCooke) [inline apollo-router-core/Cargo.toml:38]: Let's also have the comment `# don't bump to 0.3 until we bump tracing-opentelemetry to 0.16` here as well, it is super useful in the other location.

**Review by o0Ignition0o — APPROVED** [review]
LGTM.

I'll try to figure out why the span is manually created once we have tests.

File: apollo-router-core/Cargo.toml (hunk 1 above)
Comment (Geal) [reply apollo-router-core/Cargo.toml:39]: done in c3aac0e

**Comment by Geal** [comment]
FYI: https://github.com/TrueLayer/reqwest-middleware/pull/18

File: apollo-router/src/apollo_router.rs [hunk 2]
```diff
@@ -98,13 +100,17 @@ impl PreparedQuery for ApolloPreparedQuery {
                         Arc::clone(&request),
                         Arc::clone(&self.service_registry),
                         Arc::clone(&self.schema),
-                    );
-                let query_task = self.query_cache.get_query(&request.query);
+                    )
+                    .instrument(tracing::info_span!(parent: &span, "execution"));
```
[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:104]: The attribute does not allow overriding the parent span: https://docs.rs/tracing/0.1.29/tracing/attr.instrument.html

File: apollo-router/src/apollo_router.rs [hunk 3]
```diff
@@ -88,6 +89,7 @@ pub struct ApolloPreparedQuery {
 impl PreparedQuery for ApolloPreparedQuery {
     #[tracing::instrument(level = "debug")]
     async fn execute(self, request: Arc<Request>) -> ResponseStream {
+        let span = Span::current();
         stream::once(
             async move {
```
[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:94]: Yep, I tried that, but it makes the future non Send. I think I found another way without setting the parent explicitely, I'll try this afternoon

File: apollo-router/src/apollo_router.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cecton) [reply apollo-router/src/apollo_router.rs:104]: Yes this comment was not about overriding parent span and stuff. More a question of:
1. where should go the trace: at the source or at the caller
2. is the decorator more esthetic, should we use it, can we

[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:104]: that  topic should go in https://github.com/apollographql/router/pull/213

[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:104]: I'm fine either way, as long as we make sure we don't serialize too much info with the attribute

File: apollo-router/src/apollo_router.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:94]: I should have thought of that earlier :sweat_smile:  https://github.com/apollographql/router/pull/224/commits/23020aec959857b5dc90d568e49a1913854ba53a

Maybe we can simplify `ResponseStream` now

[the remark this answers is not in the document]
Comment (cecton) [reply apollo-router/src/apollo_router.rs:94]: Cool! ^_^ some improvements coming then? I'm glad I looked at this PR even if it was a bit late

[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/apollo_router.rs:94]: There's always room for improvement 😀

**Pull request merged**